### DISCUSSION TITLE
Disable "InplaceAdd" button

### DISCUSSION THREAD

**Discussion by patriciomh1965** [serenity-is/Serenity#5713] [Help]
Hello Community
Does anyone know how to disable the "Inplace Add" button according to a condition in read mode. According to the registry it can be modified or read.
I attach an image and part of the code created:

![panta_InplaceAdd](https://user-images.githubusercontent.com/42218266/118073655-49c59680-b37a-11eb-883c-b3c685b8c953.png)

XYZRow
.....
.....
[DisplayName("Compañia")]
        [Origin("cia"), MinSelectLevel(SelectLevel.List)]
        public String CiaGlosa
        {
            get { return Fields.CiaGlosa[this]; }
            set { Fields.CiaGlosa[this] = value; }
        }
        //     
        [DisplayName("Poliza"), Column("pln_poliza"), ForeignKey(typeof(TaCncConfigPolizaNuevaRow)), LeftJoin("pol"), TextualField("PolOrigenPoliza")]
        [LookupEditor(typeof(TaCncConfigPolizaNuevaRow), CascadeFrom = "PlnCia", CascadeField = "PolCia", InplaceAdd = true)]
        public Int32? PolNumero
        {
            get { return Fields.PolNumero[this]; }
            set { Fields.PolNumero[this] = value; }
        }
        [DisplayName("Poliza")]
        [Origin("pol"), MinSelectLevel(SelectLevel.List)]
        public String PolOrigenPoliza
        {
            get { return Fields.PolOrigenPoliza[this]; }
            set { Fields.PolOrigenPoliza[this] = value; }
        }

.....
......

XYZDialog

 protected afterLoadEntity() {
            super.afterLoadEntity();

            Serenity.EditorUtils.setReadOnly(this.form.PlnPlataforma, true);
            Serenity.EditorUtils.setReadOnly(this.form.PprCodPlanProd, true);
            Serenity.EditorUtils.setReadOnly(this.form.PlnCodPlanProd, true);
            Serenity.EditorUtils.setReadOnly(this.form.TprCodigoOrigen, true);
            Serenity.EditorUtils.setReadOnly(this.form.RamCodigoOrigen, true);
            Serenity.EditorUtils.setReadOnly(this.form.TseCodigoOrigen, true);
            Serenity.EditorUtils.setReadOnly(this.form.PrdCodigoOrigen, true);
            Serenity.EditorUtils.setReadOnly(this.form.PlnOrigen, true);
            
            // plan en proceso
            if (parseInt(this.form.PlnPrcEstado.value) == 2) {
                this.toolbar.findButton('apply-changes-button').toggleClass('disabled', true);
                this.toolbar.findButton('delete-button').toggleClass('disabled', true);
                this.toolbar.findButton('save-and-close-button').toggleClass('disabled', true);
                this.toolbar.findButton('grabar-button').toggleClass('disabled', true);

                Serenity.EditorUtils.setReadOnly(this.form.TseNumero, true)
                Serenity.EditorUtils.setReadOnly(this.form.RamNumero, true)
                Serenity.EditorUtils.setReadOnly(this.form.PrdNumero, true)
                Serenity.EditorUtils.setReadOnly(this.form.TprNumero, true)
                Serenity.EditorUtils.setReadOnly(this.form.PprNumero, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnOrigen, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnCia, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnMontoAsegurado, true)
                Serenity.EditorUtils.setReadOnly(this.form.PolNumero, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnDescripcion, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnDescripcionLarga, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnMinAsegurado, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnMaxAsegurado, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnMaxMayorAseg, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnCodigoTipoPlan, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnCodigoMoneda, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnIva, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnIvaOrigen, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnCoberturaExenta, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnSegmento, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnTipoDocto, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnRubro, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnMontoAsegurado, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnModRecaudacion, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnNropolcol, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnNumPolCia, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnPlataformaVenta, true)
                Serenity.EditorUtils.setReadOnly(this.form.PlnGlosaProductoOrigen, true)                
                Serenity.EditorUtils.setReadOnly(this.form.PlnEstado, true)
            }
            else {
                // Pendiente o Aplicado
                if (parseInt(this.form.PlnPrcEstado.value) == 0 || parseInt(this.form.PlnPrcEstado.value) == 1) {
                    // PLAN ISOL
                    if (parseInt(this.form.PlnOrigen.value) == 2) {
                        Serenity.EditorUtils.setReadOnly(this.form.PlnOrigen, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnCia, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnMontoAsegurado, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PolNumero, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnDescripcion, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnDescripcionLarga, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnMinAsegurado, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnMaxAsegurado, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnMaxMayorAseg, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnCodigoTipoPlan, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnCodigoMoneda, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnIva, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnIvaOrigen, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnCoberturaExenta, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnSegmento, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnTipoDocto, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnRubro, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnMontoAsegurado, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnModRecaudacion, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnNropolcol, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnNumPolCia, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnPlataformaVenta, true)
                        Serenity.EditorUtils.setReadOnly(this.form.PlnEstado, true)
                    }
                }
            }

......
.....

Since I appreciate your help

Atte.

**Comment by VictorTomaili** [comment]
While this dialog is readonly, can you navigate to "Poliza" page and edit this record? If you can, you also must can edit with inplace button. 
You also can just find the button with jquery & selectors then disable it.
